test/periphery/integration/merkle-campaign/shared/clawback/clawback.t.sol

```diff
@@ -21,6 +18,15 @@ abstract contract Clawback_Integration_Test is MerkleCampaign_Integration_Shared
         test_Clawback(users.campaignOwner);
     }
 
+    modifier whenFirstClaimMade() {
```

Comment:
What do you think about the following:
1. Move this modifier into `Modifiers`
2. Define a virtual `claim` function in `Modifiers`
3. Override `claim` through `MerkleCampaign_Integration_Test`

This will allow us to move all modifiers into `Modifiers`. Having a virtual `claim` function in Modifiers shouldn't be an issue. We already use similar technique in Core tests with create stream functions.

Reply:
it’s an interesting idea; however, since this is the only place where this modifier is being used, i would keep it here.

> Modifiers shouldn't be an issue. We already use similar technique in Core tests with create stream functions

this isn’t quite the same thing, as in core we have create virtual functions that don’t exist in the base contract (`SablierLockup`). here, `claim` is actually declared in the merkle base contract

Reply:
- Those virtual functions in core that don't exist in base contract. And then they are declared in LL/LD/LT test contracts.
- `claim` does not exist in the base contract. And `claim` is declared in MerkleCampaign contract. 

How are these two not same things? Also even if its just one Modifier, why make it an exception when we can keep all of them in the same place called `Modifiers`?

Reply:
> * Those virtual functions in core that don't exist in base contract. And then they are declared in LL/LD/LT test contracts

they are declared as _virtual_ in the `Integration_Test` which +/- the same thing with `Base_Test`

> How are these two not same things? 

`LL is BaseLockup`

`create` functions do not exist in `BaseLockup` but is fully implemented `LL`

`MerkleLL is MerkleBase`
 
`claim` function exists in `MerkleBase` and it just partially implemented in `MerkleLL` (as an internal function

so that we can create a default stream in `lockup` (shared dir) we "forced" to declare the function as virtual (as we can't call `lockup.create`

> There are a few more modifiers that are also only used once yet we have them in the Modifiers contract

most likely they don't have unique logic and are empty modifiers, if you have strong argument, feel free to push this change

Reply:
Lets move forward with your approach then. My main motivation was to move "all" modifiers into one contract regardless of what logic they contain inside.